pkg/config/config.go

```diff
@@ -1,155 +1,211 @@
 package config
 
 import (
-	"regexp"
-)
+	"strings"
 
-const Version = "v1beta1"
+	"github.com/loft-sh/vcluster/config"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/discovery"
+	"k8s.io/klog/v2"
+)
 
-type Config struct {
-	// Version is the config version
-	Version string `json:"version,omitempty" yaml:"version,omitempty"`
+// VirtualClusterConfig wraps the config and adds extra info such as name, serviceName and targetNamespace
+type VirtualClusterConfig struct {
+	// Holds the vCluster config
+	config.Config `json:",inline"`
 
-	// Exports syncs a resource from the virtual cluster to the host
-	Exports []*Export `json:"export,omitempty" yaml:"export,omitempty"`
+	// Name is the name of the vCluster
+	Name string `json:"name,omitempty"`
 
-	// Imports syncs a resource from the host cluster to virtual cluster
-	Imports []*Import `json:"import,omitempty" yaml:"import,omitempty"`
+	// ServiceName is the name of the service of the vCluster
+	ServiceName string `json:"serviceName,omitempty"`
 
-	// Hooks are hooks that can be used to inject custom patches before syncing
-	Hooks *Hooks `json:"hooks,omitempty" yaml:"hooks,omitempty"`
+	// TargetNamespace is the namespace where the workloads go
+	TargetNamespace string `json:"targetNamespace,omitempty"`
 }
 
-type Hooks struct {
-	// HostToVirtual is a hook that is executed before syncing from the host to the virtual cluster
-	HostToVirtual []*Hook `json:"hostToVirtual,omitempty" yaml:"hostToVirtual,omitempty"`
-
-	// VirtualToHost is a hook that is executed before syncing from the virtual to the host cluster
-	VirtualToHost []*Hook `json:"virtualToHost,omitempty" yaml:"virtualToHost,omitempty"`
+func (v VirtualClusterConfig) Distro() string {
+	if v.Config.ControlPlane.Distro.K3S.Enabled {
+		return config.K3SDistro
+	} else if v.Config.ControlPlane.Distro.K0S.Enabled {
```

Comment:
this is a little redundant no? without that `if` the function is the same

Reply:
Thing is more that we want to have k3s is all are disabled, so we need that function

Reply:
yeah but it's not the point, if you lose 
```go
	if v.Config.ControlPlane.Distro.K3S.Enabled {
		return config.K3SDistro
	} else
``` 
the end result is the same still. because the default will be to return k3s